### PR TITLE
🚸 [404 page] Update error page layout

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2022-12-21 下午8 27 21

### CODE REVIEW

File: src/locales/en.json [hunk 1]
```diff
@@ -26,9 +26,16 @@
   "campaign_nft_upcoming": "Next Writing NFT coming on",
   "campaign_nft_view_nft_dashboard": "Visit NFT Dashboard",
   "clipboard_copied": "Copied",
-  "dashboard_title": "My Dashbaord",
+  "dashboard_title": "My Dashboard",
   "dashboard_description": "Connect your wallet to view your NFT dashboard.",
   "dashboard_portfolio_button": "Public Portfolio",
+  "error_page_not_found_return_back":"Return Back",
```
Comment (williamchong) [inline src/locales/en.json:32]:
```suggestion
  "error_page_not_found_return_back":"Go Back",
```

File: src/locales/en.json [hunk 2]
```diff
@@ -26,9 +26,16 @@
...
+  "error_page_not_found_return_home":"Return Home",
+  "error_page_not_found_report_problem":"Report Problem",
+  "error_page_not_found_title":"Sorry, we could not find that page",
+  "error_page_not_found_title_widget":"You have found a page that does not exist",
+  "error_page_not_found_description":"If you think something is broken, report a problem.",
+  "error_page_not_found_description_widget":"Breathe in, and on the out breathe, go back and try again.",
```
Comment (williamchong) [inline src/locales/en.json:38]:
```suggestion
  "error_page_not_found_description_widget":"Breathe in, and on the out breath, go back and try again.",
```

File: src/locales/en.json [hunk 3]
```diff
@@ -26,9 +26,16 @@
...
-  "dashboard_title": "My Dashbaord",
+  "dashboard_title": "My Dashboard",
   "dashboard_description": "Connect your wallet to view your NFT dashboard.",
   "dashboard_portfolio_button": "Public Portfolio",
+  "error_page_not_found_return_back":"Return Back",
+  "error_page_not_found_return_home":"Return Home",
+  "error_page_not_found_report_problem":"Report Problem",
+  "error_page_not_found_title":"Sorry, we could not find that page",
+  "error_page_not_found_title_widget":"You have found a page that does not exist",
+  "error_page_not_found_description":"If you think something is broken, report a problem.",
```
Comment (williamchong) [inline src/locales/en.json:37]:
```suggestion
  "error_page_not_found_description":"If you think something is broken, report an issue.",
```

File: src/locales/en.json [hunk 4]
```diff
@@ -26,9 +26,16 @@
   "campaign_nft_upcoming": "Next Writing NFT coming on",
   "campaign_nft_view_nft_dashboard": "Visit NFT Dashboard",
   "clipboard_copied": "Copied",
-  "dashboard_title": "My Dashbaord",
+  "dashboard_title": "My Dashboard",
   "dashboard_description": "Connect your wallet to view your NFT dashboard.",
   "dashboard_portfolio_button": "Public Portfolio",
+  "error_page_not_found_return_back":"Return Back",
+  "error_page_not_found_return_home":"Return Home",
+  "error_page_not_found_report_problem":"Report Problem",
```
Comment (williamchong) [inline src/locales/en.json:34]:
```suggestion
  "error_page_not_found_report_problem":"Report Issue",
```

File: src/layouts/error.vue [hunk 5]
```diff
@@ -1,305 +1,63 @@
...
+      gap-[28px]
+      bg-shade-gray
+    "
+  >
+    <NFTWidget
+      class="shadow-lg w-full max-w-[400px]"
+      :title="$t('error_page_not_found_title_widget')"
+      :description="$t('error_page_not_found_description_widget')"
+      :price="404"
+      img-src="/images/error/pageNotFound.svg"
```
Comment (williamchong) [inline src/layouts/error.vue:20]: Just put it in `assets/` and use import, no need for static

File: src/layouts/error.vue [hunk 6]
```diff
@@ -1,305 +1,63 @@
 <template>
-  <Transition name="error-dialog-" appear="appear">
-    <DialogLayout class="error-dialog">
-      <template
-        v-if="isLoginErrorFromCivicLikerRegistration"
-        #header-content
+  <div
+    class="
+      flex flex-col
```
Comment (williamchong) [inline src/layouts/error.vue:4]:
```suggestion
      flex
      flex-col
```

File: src/layouts/error.vue [hunk 7]
```diff
@@ -1,305 +1,63 @@
...
-            </div>
-          </main>
-        </Transition>
-      </template>
-    </DialogLayout>
-  </Transition>
+        <ButtonV2
+          :text="$t('error_page_not_found_report_problem')"
+          preset="outline"
+          href="https://discord.com/channels/763001015712350231/814761730349596712"
```
Comment (williamchong) [inline src/layouts/error.vue:52]: I think we can actually trigger crisp
https://github.com/likecoin/liker-land/pull/726

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

File: src/layouts/error.vue [hunk 8]
```diff
@@ -1,305 +1,85 @@
...
+      px-[24px]
+      gap-[28px]
+      bg-shade-gray
+    "
+  >
+    <NFTWidget
+      class="shadow-lg w-full max-w-[400px]"
+      :title="$t('error_page_not_found_title_widget')"
+      :description="$t('error_page_not_found_description_widget')"
+      :price="404"
```
Comment (nwingt) [inline src/layouts/error.vue:20]: should return proper error code from `error.statusCode`?

Ref: https://nuxtjs.org/docs/directory-structure/layouts#error-page

File: src/layouts/error.vue [hunk 9]
```diff
@@ -109,197 +72,20 @@ export default {
   },
   data() {
     return {
-      referrer: undefined,
+      errorImg: pageNotFound,
```
Comment (nwingt) [inline src/layouts/error.vue:75]: Should error image respect the error code? I think showing 404 img but status code 500 is ambiguous

Comment (AuroraHuang22) [reply src/layouts/error.vue]: 截圖 2022-12-23 下午1 07 30

File: src/layouts/error.vue [hunk 10]
```diff
@@ -1,305 +1,101 @@
...
+      :price="error.statusCode || 404"
+      :img-src="errorImg"
+      :collect-button-label="$t('error_page_not_found_return_back')"
+      @collect="() => $router.go(-1)"
+    />
+    <div class="flex flex-col">
+      <Label
+        preset="h2"
+        class="text-like-green"
+        :text="$t('error_page_not_found_title')"
```
Comment (nwingt) [inline src/layouts/error.vue:29]: Please show error message respect to error code

**Review by nwingt — APPROVED** [review]

File: src/layouts/error.vue (hunk 10 above)
Comment (AuroraHuang22) [reply src/layouts/error.vue]: 截圖 2022-12-23 下午3 19 47

File: src/layouts/error.vue [hunk 11]
```diff
@@ -1,305 +1,118 @@
...
-            route: {
-              name,
-              params,
-              query,
-              hash,
-            },
-            ts: Date.now(),
-          })
-        );
+        default:
```
Comment (nwingt) [inline src/layouts/error.vue:94]: Seem default same as case 4?
```suggestion
        case '4':
        default:
```

Comment (AuroraHuang22) [reply src/layouts/error.vue:87]: oh!

**Review by nwingt — COMMENTED** [review]
Can src/assets/images/error/pageNotFound.png be removed?

File: src/locales/en.json [hunk 12]
```diff
@@ -26,9 +26,18 @@
...
   "dashboard_portfolio_button": "Public Portfolio",
+  "error_page_not_found_return_back":"Go Back",
+  "error_page_not_found_return_home":"Return Home",
+  "error_page_not_found_report_problem":"Report Issue",
+  "error_page_not_found_title_404":"Sorry, we could not find that page",
+  "error_page_not_found_title_500":"Well, this is unexpected",
+  "error_page_not_found_title_widget":"You have found a page that does not exist",
+  "error_page_not_found_description_404":"If you think something is broken, report an issue.",
+  "error_page_not_found_description_500":"Things are a little unstable here. We will be up and running shortly.",
+  "error_page_not_found_description_widget":"Breathe in, and on the out breath, go back and try again.",
```
Comment (nwingt) [inline src/locales/en.json:40]:
```suggestion
  "error_page_not_found_return_back": "Go Back",
  "error_page_not_found_return_home": "Return Home",
  "error_page_not_found_report_problem": "Report Issue",
  "error_page_not_found_title_404": "Sorry, we could not find that page",
  "error_page_not_found_title_500": "Well, this is unexpected",
  "error_page_not_found_title_widget": "You have found a page that does not exist",
  "error_page_not_found_description_404": "If you think something is broken, report an issue.",
  "error_page_not_found_description_500": "Things are a little unstable here. We will be up and running shortly.",
  "error_page_not_found_description_widget": "Breathe in, and on the out breath, go back and try again.",
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/layouts/error.vue [hunk 13]
```diff
@@ -1,305 +1,111 @@
...
-} from '~/util/api';
-import { getAvatarHaloTypeFromUser, checkUserNameValid } from '~/util/user';
-import { logTrackerEvent } from '~/util/EventLogger';
-import { defaultLocale } from '~/locales';
-
-import CrispMixin from '~/mixins/crisp';
-
-import DialogLayout from '~/components/DialogLayout';
-import LikeButtonAnimation from '~/components/LikeButtonAnimation';
+import pageNotFound from '~/assets/images/error/404.svg';
```
Comment (nwingt) [inline src/layouts/error.vue:61]: Can `src/assets/images/error/pageNotFound.png` be removed?

**Review by nwingt — APPROVED** [review]
Merry Christmas 🎄

**Pull request merged**